src/unit/test_ziplist.c

```diff
@@ -0,0 +1,886 @@
+
+    TEST_ASSERT(e[2].prevrawlensize == 5 && e[2].prevrawlen == strEntryBytesLarge(s2));
+    TEST_ASSERT(compareHelper(zl, 'b', s1, 2));
+
+    TEST_ASSERT(e[3].prevrawlensize == 5 && e[3].prevrawlen == strEntryBytesLarge(s1));
+    TEST_ASSERT(compareHelper(zl, 'a', s1, 3));
+
+
+    /* Delete from mid. */
+    unsigned char *p = ziplistIndex(zl, 2);
+    zl = ziplistDelete(zl, &p);
+    verify(zl, e);
+
+    TEST_ASSERT(e[0].prevrawlensize == 1 && e[0].prevrawlen == 0);
+    TEST_ASSERT(compareHelper(zl, 'd', s2, 0));
+
+    TEST_ASSERT(e[1].prevrawlensize == 5 && e[1].prevrawlen == strEntryBytesSmall(s2));
+    TEST_ASSERT(compareHelper(zl, 'c', s2, 1));
+
+    TEST_ASSERT(e[2].prevrawlensize == 5 && e[2].prevrawlen == strEntryBytesLarge(s2));
+    TEST_ASSERT(compareHelper(zl, 'a', s1, 2));
+
+
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistInsertEdgeCase(int argc, char **argv, int flags) {
+    UNUSED(flags);
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    //From issue #7170
+    zl = ziplistNew();
+
+    /* We set some values to almost reach the critical point - 254 */
+    char A_252[253] = {0}, A_250[251] = {0};
+    memset(A_252, 'A', 252);
+    memset(A_250, 'A', 250);
+
+    /* After the rpush, the list look like: [one two A_252 A_250 three 10] */
+    zl = ziplistPush(zl, (unsigned char *) "one", 3, ZIPLIST_TAIL);
+    zl = ziplistPush(zl, (unsigned char *) "two", 3, ZIPLIST_TAIL);
+    zl = ziplistPush(zl, (unsigned char *) A_252, strlen(A_252), ZIPLIST_TAIL);
+    zl = ziplistPush(zl, (unsigned char *) A_250, strlen(A_250), ZIPLIST_TAIL);
+    zl = ziplistPush(zl, (unsigned char *) "three", 5, ZIPLIST_TAIL);
+    zl = ziplistPush(zl, (unsigned char *) "10", 2, ZIPLIST_TAIL);
+
+    p = ziplistIndex(zl, 2);
+    TEST_ASSERT(ziplistCompare(p, (unsigned char *) A_252, strlen(A_252)));
+
+    /* When we remove A_252, the list became: [one two A_250 three 10]
+     * A_250's prev node became node two, because node two quite small
+     * So A_250's prevlenSize shrink to 1, A_250's total size became 253(1+2+250)
+     * The prev node of node three is still node A_250.
+     * We will not shrink the node three's prevlenSize, keep it at 5 bytes */
+    zl = ziplistDelete(zl, &p);
+
+    p = ziplistIndex(zl, 3);
+    TEST_ASSERT(ziplistCompare(p, (unsigned char *) "three", 5));
+
+    /* We want to insert a node after A_250, the list became: [one two A_250 10 three 10]
+     * Because the new node is quite small, node three prevlenSize will shrink to 1 */
+    zl = ziplistInsert(zl, p, (unsigned char *) "10", 2);
+
+    /* Last element should equal 10 */
+    p = ziplistIndex(zl, -1);
+    TEST_ASSERT(ziplistCompare(p, (unsigned char *) "10", 2));
+
+    zfree(zl);
+    return 0;
+}
+
+int test_ziplistBenchmarks(int argc, char **argv, int flags) {
+    if (argc >= 4)
+        srand(atoi(argv[3]));
+    int accurate = (flags & UNIT_TEST_ACCURATE);
+    printf("Stress with variable ziplist size:\n");
```

Comment:
```suggestion
    TEST_PRINT_INFO("Stress with variable ziplist size:");
```
Ideally let's omit any direct printf call. We can better capture it in the future this way.